Copy edit 5.16 API articles, A through F (#1993)
sensu/sensu-docs: content/sensu-go/5.16/api/filters.md

---
title: "Filters API"

> description: "The Sensu filters API provides HTTP access to filter data. This reference includes examples for returning lists of filters, creating Sensu filters, and more. Read on for the full reference."

Reviewer: ```suggestion
description: "The Sensu filters API provides HTTP access to event filter data. This reference includes examples for returning lists of filters, creating Sensu filters, and more. Read on for the full reference."
```

#### EXAMPLE {#filters-get-example}

> The following example demonstrates a request to the `/filters` API, resulting in a JSON array that contains [filter definitions][1].

Reviewer: ```suggestion
The following example demonstrates a request to the `/filters` API, resulting in a JSON array that contains [event filter definitions][1].
```

### `/filters` (POST)

> The `/filters` API endpoint provides HTTP POST access to create a filter.

Reviewer: ```suggestion
The `/filters` API endpoint provides HTTP POST access to create an event filter.
```

#### API Specification {#filters-post-specification}

/filters (POST) | 
----------------|------

> description     | Creates a Sensu filter.

Reviewer: ```suggestion
description     | Creates a Sensu event filter.
```

/filters/:filter (GET) | 
---------------------|------

> description          | Returns the specified filter.

Reviewer: ```suggestion
description          | Returns the specified event filter.
```

### `/filters/:filter` (PUT) {#filtersfilter-put}

> The `/filters/:filter` API endpoint provides HTTP PUT access to create or update a filter.

Reviewer: ```suggestion
The `/filters/:filter` API endpoint provides HTTP PUT access to create or update an event filter.
```

#### API Specification {#filtersfilter-put-specification}

/filters/:filter (PUT) | 
----------------|------

> description     | Creates or updates the specified Sensu filter.

Reviewer: ```suggestion
description     | Creates or updates the specified Sensu event filter.
```

### `/filters/:filter` (DELETE) {#filtersfilter-delete}

> The `/filters/:filter` API endpoint provides HTTP DELETE access to delete a filter from Sensu (specified by the filter name).

Reviewer: ```suggestion
The `/filters/:filter` API endpoint provides HTTP DELETE access to delete an event filter from Sensu (specified by the filter name).
```

#### EXAMPLE {#filtersfilter-delete-example}

> The following example shows a request to delete the filter `production-only`, resulting in a successful HTTP `204 No Content` response.

Reviewer: ```suggestion
The following example shows a request to delete the event filter `production-only`, resulting in a successful HTTP `204 No Content` response.
```

/filters/:filter (DELETE) | 
--------------------------|------

> description               | Removes the specified filter from Sensu.

Reviewer: ```suggestion
description               | Removes the specified event filter from Sensu.
```
